lib/commands/project/project.js

```diff
@@ -363,3 +372,63 @@ function spawn(command, options, cwd, cb) {
     cb(err);
   });
 }
+
+
+//.option('-p, --path-name [path]', 'a sepecific path of the api')
+//.option('-f, --test-module <module>', 'one of: ' + testmodules)
+//.option('-t, --assertion-format <type>', 'one of: ' + assertiontypes)
+function testGenerate(directory, options, cb) {
+  var pathList = [];
+  var desiredPaths = [];
+  var testModule = options.testModule || TESTMODULES[0];
+  var assertionFormat = options.assertionFormat || ASSERTIONTYPES[0];
+
+  //read the yaml file and validate it
+  readProject(directory, options, function(err, project) {
+    if (err) { return cb(err); }
+    swaggerSpec.validateSwagger(project.api.swagger, options, function(err) {
+      // get the array of string paths from json object
+      pathList = Object.keys(project.api.swagger.paths);
+
+      //check if the test frame is one of the two
+      if (options.testModule && !_.includes(TESTMODULES, options.testModule)) {
+        return cb(new Error(util.format('Unknown type: %j. Valid types: %s', options.testModule, TESTMODULES.join(', '))));
+      }
+
+      // check if the assertion-format is one of the three
+      if (options.assertionFormat && !_.includes(ASSERTIONTYPES, options.assertionFormat)) {
+        return cb(new Error(util.format('Unknown type: %j. Valid types: %s', options.assertionFormat, ASSERTIONTYPES.join(', '))));
+      }
+
+      // process the paths option
+      if (options.pathName){
+        var reg = new RegExp(options.pathName);
+        desiredPaths = pathList.filter(function(val) {
+          return val.match(reg);
+        });
+      }
+
+      // get the result
+      var config = {
+        pathName: desiredPaths,
+        testModule: testModule,
+        assertionFormat: assertionFormat
+      };
+
+      var result = template.testGen(project.api.swagger, config);
+
+      //output the result
+      async.eachSeries(result, function(item, cb) {
```

Comment:
Hey Moshen, I think the async.eachSeries doesn't work. Do you have any tips on that?